docs/DeveloperGuide.md

```diff
@@ -173,48 +173,58 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+This will then call `Addressbook#updateAssignmentList(person)`, causing the `assignments` in `AddressBook` to be replaced with the assignments in 2nd person's assignment list.
 
-Step 2. The user inputs `show 2` command to display the 2nd person's assignment list in the address book. The `show` command will then call `Model#updateFilterdAssignmentList(person)`, whereby `person` variable is the 2nd person in the address book.
-This causes the `assignments` in `AddressBook` to be replaced with the 2nd person's assignment list.
+The two object diagram below shows illustrates how the objects interacts and changes when a `show` command is executed.
 
-![AssignmentState1](images/AssignmentState1.png)
+![DisplayAssignmentObjectDiagram1](images/DisplayAssignmentObjectDiagram1.png))
+![DisplayAssignmentObjectDiagram2](images/DisplayAssignmentObjectDiagram2.png))
 
+Step 3. When `assignments` is updated, it is retrieved by the `Logic` using `Model#getAssignmentList()` to input into the assignment list panel of the `UI`
+This results in the assignment list panel to display the assignments of the 2nd person.
 
-Step 3. When `assignments` is updated, it is retrieved by the `Logic` using `Model#getFilteredAssignmentList()` to input into the assignment panel of the `Ui`
-This results in the assignment list panel to display the assignments of the person.
+Step 4. The user decides to modify the assignment list of the person by using either `give`, `done` or `remove` command. This will result in the assignment list in the person to be modified.
+The command will then call `Model#updateAssignmentList(person)` to get the recent updated assignment list to replace `assignments`.
 
-![AssignmentState2](images/AssignmentState2.png)
+Step 5. Step 3 is repeated to display the recent updated assignment list.
 
+The sequence diagram below illustrates the interactions between the `Logic` and `Model` component, when an assignment command (e.g `show`, `give`, `done`, `remove`) is called.
 
-Step 4. The user decides to modify the assignment list of the person by using either `give`, `done` or `remove` command. This will result in the assignment list in the person to be modified.
-The command will the call `Model#updateFilteredAssignmentList(person)` to get the recent updated assignment list to replace `assignments`.
+![DisplayAssignmentListSequenceDiagram](images/DisplayAssignmentListSequenceDiagram.png)
 
-![AssignmentState3](images/AssignmentState3.png)
+#### Design considerations:
 
+**Aspect: How the assignment list can be displayed:**
 
-Step 5. Step 3 is repeated to show the recent updated assignment list.
+* **Alternative 1(current choice):** Displays assignment list next to the contact list panel in the same window.
 
-![AssignmentState4](images/AssignmentState4.png)
+  * Pros: Allows for better grouping of assignment commands, so that commands are separated cleanly.<br>
```

Comment:
I think you might have mixed the explanation.

Current choice 
Pro: Allows you to do everything on one window
Cons: Commands that deal with persons and assignments need to be distinctly named as they share the same window. For example, `add` person and `add` assignment will have conflict.

Alternative choice
Pro: Allows you to cleanly segregate commands of assignments and persons because they are on different windows.
Cons: (Your cons is correct)